diff --git a/dist/cjs/asar-utils.js b/dist/cjs/asar-utils.js
index c09cee642d347f5ec59a57d56674119e6eccb911..26881c6b872ea61b77087b2ffb36f58f6e014a90 100644
--- a/dist/cjs/asar-utils.js
+++ b/dist/cjs/asar-utils.js
@@ -155,7 +155,12 @@ const mergeASARs = async ({ x64AsarPath, arm64AsarPath, outputAsarPath, singleAr
             (0, child_process_1.execFileSync)(LIPO, [source, destination, '-create', '-output', destination]);
         }
         (0, debug_1.d)(`creating archive at ${outputAsarPath}`);
-        const resolvedUnpack = Array.from(unpackedFiles).map((file) => path_1.default.join(x64Dir, file));
+        const unpackDir = unpackedFiles.size > 0 && Array.from(unpackedFiles).every((file) => file === 'node_modules' || file.startsWith(`node_modules${path_1.default.sep}`))
+            ? 'node_modules'
+            : undefined;
+        const resolvedUnpack = unpackDir
+            ? []
+            : Array.from(unpackedFiles).map((file) => path_1.default.join(x64Dir, file));
         let unpack;
         if (resolvedUnpack.length > 1) {
             unpack = `{${resolvedUnpack.join(',')}}`;
@@ -165,6 +170,7 @@ const mergeASARs = async ({ x64AsarPath, arm64AsarPath, outputAsarPath, singleAr
         }
         await asar_1.default.createPackageWithOptions(x64Dir, outputAsarPath, {
             unpack,
+            unpackDir,
         });
         (0, debug_1.d)('done merging');
     }
diff --git a/dist/esm/asar-utils.js b/dist/esm/asar-utils.js
index d863cf62d197855692935b0b635d18cdcbef1a62..70ca09c5537a5a1c63921d498a3cec874b36369a 100644
--- a/dist/esm/asar-utils.js
+++ b/dist/esm/asar-utils.js
@@ -147,7 +147,12 @@ export const mergeASARs = async ({ x64AsarPath, arm64AsarPath, outputAsarPath, s
             execFileSync(LIPO, [source, destination, '-create', '-output', destination]);
         }
         d(`creating archive at ${outputAsarPath}`);
-        const resolvedUnpack = Array.from(unpackedFiles).map((file) => path.join(x64Dir, file));
+        const unpackDir = unpackedFiles.size > 0 && Array.from(unpackedFiles).every((file) => file === 'node_modules' || file.startsWith(`node_modules${path.sep}`))
+            ? 'node_modules'
+            : undefined;
+        const resolvedUnpack = unpackDir
+            ? []
+            : Array.from(unpackedFiles).map((file) => path.join(x64Dir, file));
         let unpack;
         if (resolvedUnpack.length > 1) {
             unpack = `{${resolvedUnpack.join(',')}}`;
@@ -157,6 +162,7 @@ export const mergeASARs = async ({ x64AsarPath, arm64AsarPath, outputAsarPath, s
         }
         await asar.createPackageWithOptions(x64Dir, outputAsarPath, {
             unpack,
+            unpackDir,
         });
         d('done merging');
     }
